pkg/userdata/ubuntu/testdata/cri-o-1.9-digitalocean.golden

```diff
@@ -0,0 +1,221 @@
+    Requires=network-online.target
+    After=network-online.target
+
+    [Service]
+    Type=oneshot
+    RemainAfterExit=true
+    ExecStart=/usr/local/bin/download-kubeadm
+
+- path: "/etc/systemd/system/cni-binary.service"
+  content: |
+    [Unit]
+    Requires=network-online.target
+    After=network-online.target
+
+    [Service]
+    Type=oneshot
+    RemainAfterExit=true
+    ExecStart=/usr/local/bin/download-cni
+
+- path: "/etc/systemd/system/kubeadm-unitfile.service"
+  content: |
+    [Unit]
+    Requires=network-online.target
+    After=network-online.target
+
+    [Service]
+    Type=oneshot
+    RemainAfterExit=true
+    ExecStart=/usr/local/bin/download-kubelet-kubeadm-unitfile
+
+- path: "/etc/systemd/system/kubeadm-join.service"
+  content: |
+    [Unit]
+    Requires=network-online.target
+    Requires=kubelet-binary.service kubeadm-binary.service cni-binary.service kubeadm-unitfile.service
+    After=network-online.target
+    After=kubelet-binary.service kubeadm-binary.service cni-binary.service kubeadm-unitfile.service
+    Requires=crio.service
+    After=crio.service
+
+
+    [Service]
+    Type=oneshot
+    RemainAfterExit=true
+    Environment="PATH=/sbin:/bin:/usr/sbin:/usr/bin:/opt/bin"
+    ExecStartPre=/sbin/modprobe br_netfilter
+    ExecStart=/opt/bin/kubeadm join \
+      --cri-socket /var/run/crio/crio.sock \
+      --token my-token \
+      --discovery-token-ca-cert-hash sha256:6caecce9fedcb55d4953d61a27dc6997361a2f226ad86d7e6004dde7526fc4b1 \
+      --ignore-preflight-errors=Port-10250 \
+      server:443
+
+- path: "/etc/systemd/system/kubelet.service.d/20-extra.conf"
+  content: |
+    [Service]
+    Environment="KUBELET_EXTRA_ARGS= \
+       --container-runtime=remote --container-runtime-endpoint=unix:///var/run/crio/crio.sock --cgroup-driver=systemd"
+
+- path: "/etc/systemd/system/kubelet.service.d/30-clusterdns.conf"
+  content: |
+    [Service]
+    Environment="KUBELET_DNS_ARGS=--cluster-dns=10.10.10.10 --cluster-domain=cluster.local"
+
+- path: "/etc/systemd/system/kubelet.service"
+  content: |
+    [Unit]
+    Description=Kubelet
+    Requires=network-online.target
+    After=network-online.target
+    Requires=kubelet-binary.service kubeadm-binary.service cni-binary.service kubeadm-unitfile.service
+    After=kubelet-binary.service kubeadm-binary.service cni-binary.service kubeadm-unitfile.service
+    Requires=crio.service
+    After=crio.service
+
+    [Service]
+    Environment="PATH=/sbin:/bin:/usr/sbin:/usr/bin:/opt/bin"
+    ExecStart=/opt/bin/kubelet
+    Restart=always
+    StartLimitInterval=0
+    RestartSec=10
+    Restart=always
+
+    [Install]
+    WantedBy=multi-user.target
+- path: "/etc/sysconfig/crio-network"
+  content: |
+    CRIO_NETWORK_OPTIONS="--registry=docker.io"
+
+runcmd:
+# Required for Hetzner, because they set some arbitrary password
+# if the sshkey wasnt set via their API and require as to change
+# that password on first login, which we cant do since we dont know
```

Comment:
Then we should maybe - like with digitalocean - generate a random key for each machine we use for the API call?

Reply:
That is actually a really smart idea, will do it in a follow-up